### ISSUE TITLE
GameUserSettings.init being overwritten when restarting the container

### ISSUE THREAD

**Issue by TheMath123** [SickHub/arkserver#22]
## Image Setup Details

* **Image Tag**: `drpsychick/arkserver:latest`
* **Docker Host OS**: Linux (Ubuntu 22.04)
* **`docker run` command or `docker-compose.yml` file**: 
```yaml
services:
  ark:
    container_name: "ark"
    restart: unless-stopped
    image: drpsychick/arkserver:latest
    tty: true
    stdin_open: true
    ports:
      - "27015:27015/udp"
      - "7778:7778/udp"
      - "7777:7777/udp"
      - "32330:32330"
    environment:
      - am_ark_SessionName=My Server
      - am_serverMap=Ragnarok
      - am_ark_ServerPassword=xxxx
      - am_ark_ServerAdminPassword=xxxx
      - am_ark_MaxPlayers=70
      - am_ark_QueryPort=27015
      - am_ark_Port=7778
      - am_ark_RCONPort=32330
      - am_ark_AltSaveDirectoryName=SavedArks
      - am_arkwarnminutes=15
      - am_arkAutoUpdateOnStart=true
      - am_arkflag_NoTransferFromFiltering=
      - am_arkflag_ForceAllowCaveFlyers=
      - am_arkflag_crossplay=true
      - am_arkopt_clusterid=xxxx
      - VALIDATE_SAVE_EXISTS=false
      - BACKUP_ONSTART=false
      - LOG_RCONCHAT=0
      - ARKCLUSTER=false
      - am_ark_GameModIds=895711211,761535755,731604991,520879363,760561402,1609138312,889745138,2769908652,1650069767,1284896962,2135624159,834667662,1837445660,821530042,2559520609
    volumes:
      - ark_steamapps:/home/steam/.steam/steamapps
      - ark_data:/ark
      - ark_server:/arkserver
      - ark_saves:/arkserver/ShooterGame/Saved
      - ark_clusters:/arkserver/ShooterGame/Saved/clusters
    ulimits:
      nofile:
        soft: 1000000
        hard: 1000000

volumes:
  ark_steamapps:
  ark_data:
  ark_server:
  ark_saves:
  ark_clusters:
```

## Description of Issue
I am experiencing an issue when trying to modify the game settings in the `GameUserSettings.ini` file. I edited the file located at `ark/config/server/ShooterGame/Saved/Config/Linux/GameUserSettings.ini` to adjust parameters like dinosaur taming speed, mods, and other settings. However, every time I restart the Docker container, the settings are reset to their default values.

## Additional Information
It seems that changes made to the `GameUserSettings.ini` file are not being persisted. I would like a solution to ensure the settings remain intact after restarting the container.

**Comment by DrPsychick** [comment]
The image uses `arkmanager` to configure and start ARK. You could try to apply the settings through arkmanager: https://github.com/arkmanager/ark-server-tools?tab=readme-ov-file#instance-config-files

Apart from that, there is also an option to provide your own `GameUserSettings.ini` with `arkGameUserSettingsIniFile=path/to/GameUserSettings.ini`, see also https://github.com/arkmanager/ark-server-tools/blob/master/tools/arkmanager.cfg.shellcheck

I hope that helps.

**Comment by TheMath123** [comment]
I created a file called main.GameUserSettings.ini and placed it at this path:
/home/docker/docker-storage/volumes/ark_ark_data/_data/server/ShooterGame/Saved/Config/LinuxServer. This is inside the volume used by the container.

Currently, the path where GameUserSettings.ini is being loaded on the server is the same one I’m trying to modify, but it gets overwritten.

In the Docker Compose file, I added the following environment variable:
am_arkGameUserSettingsIniFile=/ark/server/ShooterGame/Saved/Config/LinuxServer/main.GameUserSettings.ini

However, it’s not being applied, and I’m unsure where the custom config file should be placed and how it should be declared in the environment variables.

Can you provide a practical example of how to do this? I’ve tried several approaches, but I still haven’t succeeded.

**Comment by TheMath123** [comment]
I managed to make it work. Basically, I added the following to the `docker-compose.yml` file:  
```yml
am_arkGameUserSettingsIniFile=/ark/server/ShooterGame/Saved/Config/LinuxServer/main.GameUserSettings.ini
```  

With the custom configuration file `main.GameUserSettings.ini` placed in the Docker volume at:  
```
/home/docker/docker-storage/volumes/ark_ark_data/_data/server/ShooterGame/Saved/Config/LinuxServer
```  

This way, we were able to customize the settings, and it worked correctly.  

---

My friend, who uses a shared ARK account on Steam, wasn’t able to access the server without enabling the following setting:  
```
AllowSharedConnections=true
```  
Now he can access and play.  

---

Thank you, and I hope this helps anyone with questions about this configuration.
